src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -289,7 +292,11 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                             if (refreshAll)
                             {
                                 // Trigger a single load-all operation if a change was detected in one or more key-values with refreshAll: true
-                                applicationSettings = await LoadAll(client, cancellationToken).ConfigureAwait(false);
+                                // Preserve reference to loaded data with deep copy into serverData before calling mapper functions
+                                Dictionary<string, ConfigurationSetting> data = await LoadSelectedKeyValues(client, cancellationToken).ConfigureAwait(false);
+                                watchedSettings = await LoadKeyValuesRegisteredForRefresh(client, data, cancellationToken).ConfigureAwait(false);

```

Comment:
Is the change to watched settings required for Map API, or can it be fixed in a separate PR?